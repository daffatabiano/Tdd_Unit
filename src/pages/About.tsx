import Footers from '@/fragments/Footers';
import NavbarFlowbite from '@/fragments/Navbar';

export default function About() {
    return (
        <>
            <NavbarFlowbite />
            <header
                style={{ minHeight: '400px' }}
                className="jumbotron bg-info text-white rounded d-flex align-items-center jumbotron-fluid"
            >
                <div className="container ">
                    <h1 className="display-4">Hello, world!</h1>
                    <p className="lead">
                        This is About page. It is under construction.
                        <br />
                    </p>
                </div>
            </header>
            <section className="mt-5 d-flex justify-content-around align-items-center">
                <img src="https://via.placeholder.com/300" alt="" />
                <div className="w-50">
                    <p>This is About page. It is under construction.</p>
                    <button className="btn btn-primary">Button</button>
                </div>
            </section>
            <section className="mt-5 d-flex justify-content-around align-items-center">
                <div className="d-flex align-items-end flex-column w-50 justify-content-end">
                    <p>This is About page. It is under construction.</p>
                    <button className="btn btn-primary">Button</button>
                </div>
                <img src="https://via.placeholder.com/300" alt="" />
            </section>
            <Footers />
        </>
    );
}
