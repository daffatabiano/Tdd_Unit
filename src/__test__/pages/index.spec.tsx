import React from 'react';
import { render } from '@testing-library/react';
import Home from '@/pages';
import About from '@/pages/About';

describe('Homepage', () => {
    it('should render', () => {
        const { container } = render(<Home />);

        expect(container).toMatchSnapshot();
    });
});

describe('Homepage', () => {
    it('should render', () => {
        const { getByRole } = render(<Home />);

        expect(getByRole('content')).toMatchSnapshot();
    });
});

describe('HomePage', () => {
    it('should render home text', () => {
        const { getByText } = render(<Home />);
        expect(getByText('Hello World')).toBeInTheDocument();
    });
});

describe('HomePage', () => {
    it('should render home role', () => {
        const { getByRole } = render(<Home />);
        expect(getByRole('main')).toBeInTheDocument();
    });
});

describe('About Page', () => {
    it('should render about', () => {
        const { container } = render(<About />);
        expect(container).toMatchSnapshot();
    });
});
