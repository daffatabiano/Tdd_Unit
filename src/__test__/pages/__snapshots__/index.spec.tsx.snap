// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Page should render about 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg bg-body-tertiary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          height="30"
          loading="lazy"
          src="/logo-travel.png"
          width="30"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <form
          class="d-flex"
          role="search"
        >
          <input
            aria-label="Search"
            class="form-control me-2"
            placeholder="Search"
            type="search"
          />
          <button
            class="btn btn-outline-success"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </div>
  </nav>
  <header
    class="jumbotron bg-info text-white rounded d-flex align-items-center jumbotron-fluid"
    style="min-height: 400px;"
  >
    <div
      class="container "
    >
      <h1
        class="display-4"
      >
        Hello, world!
      </h1>
      <p
        class="lead"
      >
        This is About page. It is under construction.
        <br />
      </p>
    </div>
  </header>
  <section
    class="mt-5 d-flex justify-content-around align-items-center"
  >
    <img
      alt=""
      src="https://via.placeholder.com/300"
    />
    <div
      class="w-50"
    >
      <p>
        This is About page. It is under construction.
      </p>
      <button
        class="btn btn-primary"
      >
        Button
      </button>
    </div>
  </section>
  <section
    class="mt-5 d-flex justify-content-around align-items-center"
  >
    <div
      class="d-flex align-items-end flex-column w-50 justify-content-end"
    >
      <p>
        This is About page. It is under construction.
      </p>
      <button
        class="btn btn-primary"
      >
        Button
      </button>
    </div>
    <img
      alt=""
      src="https://via.placeholder.com/300"
    />
  </section>
  <footer
    class="footer bottom-0 p-10 bg-secondary text-neutral-content"
  >
    <div
      class="items-center grid-flow-col"
    >
      <p>
        Copyright © 
        2024
         - All right reserved
      </p>
    </div>
  </footer>
</div>
`;

exports[`Homepage should render 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg bg-body-tertiary"
  >
    <div
      class="container-fluid"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <img
          alt=""
          height="30"
          loading="lazy"
          src="/logo-travel.png"
          width="30"
        />
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <form
          class="d-flex"
          role="search"
        >
          <input
            aria-label="Search"
            class="form-control me-2"
            placeholder="Search"
            type="search"
          />
          <button
            class="btn btn-outline-success"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
    </div>
  </nav>
  <main
    class="className"
    role="main"
  >
    <h1>
      Hello World
    </h1>
    <section
      class="mt-5 h-100"
      role="content"
    >
      <div
        class="carousel slide"
        data-bs-ride="carousel"
        id="carouselExampleAutoplaying"
      >
        <div
          class="carousel-inner"
        >
          <div
            class="carousel-item active"
          >
            <img
              alt="..."
              class="d-block w-100"
              src="https://source.unsplash.com/random/200x200?sig=1"
            />
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="..."
              class="d-block w-100"
              src="https://source.unsplash.com/random/200x200?sig=2"
            />
          </div>
          <div
            class="carousel-item"
          >
            <img
              alt="..."
              class="d-block w-100"
              src="https://source.unsplash.com/random/200x200?sig=3"
            />
          </div>
        </div>
        <button
          class="carousel-control-prev"
          data-bs-slide="prev"
          data-bs-target="#carouselExampleAutoplaying"
          type="button"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Previous
          </span>
        </button>
        <button
          class="carousel-control-next"
          data-bs-slide="next"
          data-bs-target="#carouselExampleAutoplaying"
          type="button"
        >
          <span
            aria-hidden="true"
            class="carousel-control-prev-icon"
          />
          <span
            class="visually-hidden"
          >
            Next
          </span>
        </button>
      </div>
    </section>
  </main>
  <footer
    class="footer bottom-0 p-10 bg-secondary text-neutral-content"
  >
    <div
      class="items-center grid-flow-col"
    >
      <p>
        Copyright © 
        2024
         - All right reserved
      </p>
    </div>
  </footer>
</div>
`;

exports[`Homepage should render 2`] = `
<section
  class="mt-5 h-100"
  role="content"
>
  <div
    class="carousel slide"
    data-bs-ride="carousel"
    id="carouselExampleAutoplaying"
  >
    <div
      class="carousel-inner"
    >
      <div
        class="carousel-item active"
      >
        <img
          alt="..."
          class="d-block w-100"
          src="https://source.unsplash.com/random/200x200?sig=1"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="..."
          class="d-block w-100"
          src="https://source.unsplash.com/random/200x200?sig=2"
        />
      </div>
      <div
        class="carousel-item"
      >
        <img
          alt="..."
          class="d-block w-100"
          src="https://source.unsplash.com/random/200x200?sig=3"
        />
      </div>
    </div>
    <button
      class="carousel-control-prev"
      data-bs-slide="prev"
      data-bs-target="#carouselExampleAutoplaying"
      type="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="visually-hidden"
      >
        Previous
      </span>
    </button>
    <button
      class="carousel-control-next"
      data-bs-slide="next"
      data-bs-target="#carouselExampleAutoplaying"
      type="button"
    >
      <span
        aria-hidden="true"
        class="carousel-control-prev-icon"
      />
      <span
        class="visually-hidden"
      >
        Next
      </span>
    </button>
  </div>
</section>
`;
